lib/eph/blobLease.ts

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import * as debugModule from "debug";
+import { createBlobService, BlobService, ServiceResponse } from "azure-storage";
+import * as Constants from "../util/constants";
+import { defaultLock } from "../util/utils";
+
+const debug = debugModule("azure:event-hubs:eph:lease");
+
+export interface CreateContainerResult {
+  created: BlobService.ContainerResult;
+  details: ServiceResponse;
+}
+
+/**
+ * Describes the lease used to checkpoint the offset of data received by a receiver for a given
+ * partition in a consumer group for an EventHub.
+ */
+export interface Lease {
+  /**
+   * @property {string} [partitionId] The associated partitionId for which the lease is held.
+   */
+  partitionId?: string;
+  /**
+   * @property {string} [leaseId] The unqiue identifier for the lease. In the BlobLease this
+   * property will be set after getting a lease from the Azure Blob storage.
+   */
+  leaseId?: string;
+  /**
+   * @property {boolean} isHeld Determines whether the lease is held. Default: false.
+   */
+  isHeld: boolean;
+  /**
+   * Acquires the lease.
+   * @param {*} options Any options that need to be passed to acquire the lease.
+   * @returns {Promise<Lease>} Promise<Lease>
+   */
+  acquire(options: any): Promise<Lease>;
+  /**
+   * Renews the lease.
+   * @param {*} options Any options that need to be passed to renew the lease.
+   * @returns {Promise<Lease>} Promise<Lease>
+   */
+  renew(options: any): Promise<Lease>;
+  /**
+   * Releases the lease.
+   * @param {*} options Any options that need to be passed to release the lease.
+   * @returns {Promise<Lease>} Promise<Lease>
+   */
+  release(options: any): Promise<Lease>;
+  /**
+   * Updates the content
+   * @param {*} options Any options that need to be passed to update the blob content.
+   * @returns {Promise<Lease>} Promise<Lease>
+   */
+  updateContent(text: string, options?: any): Promise<Lease>;
+  /**
+   * Gets the content.
+   * @param {*} [options] Any options that need to be passed to get the blob content.
+   * @returns {Promise<string>} Promise<string>
+   */
+  getContent(options?: any): Promise<string>;
+}
+
+export class BlobLease implements Lease {
+
+  static notHeldError: string = "Lease not held";
+  static _beginningOfTime: string = new Date(1990, 1, 1).toUTCString();
```

Comment:
do you mean for this to be January 1, 1990? months are zero-indexed in Javascript dates so you probably want `new Date(1990, 0, 1).toUTCString()`

also nit: should this be private?